sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1545,6 +1545,203 @@ class DataFrameAggregateSuite extends QueryTest
     )
     checkAnswer(res, Row(Array(1), Array(1)))
   }
+
+  test("SPARK-16484: hll_*_agg + hll_union + hll_sketch_estimate positive tests") {
```

Comment:
I believe you are not done with the tests yet, but please let me make an observation: all positive queries below involve a grouping aggregation. I believe we should also have tests with non-grouping aggregations.

Reply:
I thought I had tested hll_union, but that one was missing. Will add more explicit tests for non-aggregate expressions, separate+downstream of aggregate functions.

Reply:
Updated tests to validate non-aggregate functions in both grouping and non-grouping expressions.

Reply:
Will take a look at them as soon as possible.